akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -11,7 +11,35 @@ import akka.annotation.ApiMayChange
 import scala.concurrent.{ ExecutionContext, Future }
 
 @ApiMayChange
-trait Projection {
+trait Projection[StreamElement] {
+
+  /**
+   * The projection group name. The name must be unique over the whole system.
+   * For instance, a 'user-view' could be the name of a projection.
+   */
+  def groupName: String
```

Comment:
It could be `name`. I chose for `groupName` to make it clear that it can have sibilings. That name is shared with other projections doing the same processing, but consuming a different 'partition'.

Reply:
I'm fine to call it `name`

Reply:
I'll make a follow up PR to rename it, and flesh out the scaladocs a little more.  It will be a good way to solidify my own understanding. :+1: